specviz/widgets/workspace.py

```diff
@@ -354,8 +366,30 @@ def load_data(self, file_path, file_loader, display=False):
         : :class:`~specviz.core.items.DataItem`
             The `DataItem` instance that has been added to the internal model.
         """
+        # In the case that the user has selected auto load, loop through every
+        # available loader and choose the one that 1) the registry identifier
+        # function allows, and 2) is the highest priority.
         try:
-            spec = Spectrum1D.read(file_path, format=file_loader)
+            try:
+                spec = Spectrum1D.read(file_path, format=file_loader)
+            except IORegistryError as e:
+                # In this case, assume that the registry has found several
+                # loaders that fit the same identifier, choose the highest
+                # priority one.
+                fmts = identify_format('read', Spectrum1D, file_path, None, [], {})
+
+                logging.warning(e)
+                logging.warning("Loaders for '%s' matched for this data set. "
+                                "Iterating based on priority."
+                                "", ', '.join(fmts))
+
+                for fmt in fmts:
```

Comment:
Just a nit, but it would be useful to have a comment here that says the formats are already in priority order. It's not entirely obvious from the code (at least without looking in `specutils`).